src/core/Akka.Streams.Tests/Dsl/FlowSelectAsyncSpec.cs

```diff
@@ -467,5 +469,36 @@ Task<int> Deferred()
                 }
             }, Materializer);
         }
+        
+        [Theory(DisplayName = "SelectAsync with restart decider should restart")]
+        [ClassData(typeof(FailingTaskData<ImmutableList<int>>))]
+        public async Task SelectAsyncFailingTaskTest(Func<ImmutableList<int>, Task<NotUsed>> mapFunc)
+        {
+            var materializer = ActorMaterializer.Create(Sys);
+        
+            var queue = Source
+                .Queue<int>(bufferSize: 5000, overflowStrategy: OverflowStrategy.DropNew)
+                .BatchWeighted(
+                    max: 100,
+                    costFunction: i => i,
+                    seed: r => ImmutableList.Create([r]),
+                    aggregate: (oldRows, i) => oldRows.Add(i))
+                .SelectAsync(
+                    parallelism: 3,
+                    asyncMapper: mapFunc)
+                .AddAttributes(ActorAttributes.CreateSupervisionStrategy(Deciders.RestartingDecider))
```

Comment:
The stream would stop if this attribute isn't declared, `SelectAsync` default decider is `Stop`.